### PR TITLE
Allow use `Each`, `Callback` and `StopOnError` as class attribute

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  | -

### CODE REVIEW

File: src/Rule/Callback.php [hunk 1]
```diff
@@ -32,6 +32,8 @@ final class Callback implements
     use SkipOnErrorTrait;
     use WhenTrait;
 
+    private ?object $objectBeingValidated = null;
```
Comment (samdark) [inline src/Rule/Callback.php:35]:
```suggestion
    private ?object $objectValidated = null;
```

File: src/Rule/Callback.php [hunk 2]
```diff
@@ -73,8 +75,17 @@ public function getMethod(): string|null
...
 
+    public function getObjectBeingValidated(): ?object
+    {
+        return $this->objectBeingValidated;
+    }
+
     public function afterInitAttribute(object $object, int $target): void
     {
+        if ($target === Attribute::TARGET_CLASS) {
+            $this->objectBeingValidated = $object;
```
Comment (samdark) [inline src/Rule/Callback.php:86]:
```suggestion
            $this->objectValidated = $object;
```

File: src/Rule/Callback.php [hunk 3]
```diff
@@ -73,8 +75,17 @@ public function getMethod(): string|null
         return $this->method;
     }
 
+    public function getObjectBeingValidated(): ?object
+    {
+        return $this->objectBeingValidated;
```
Comment (samdark) [inline src/Rule/Callback.php:80]:
```suggestion
        return $this->objectValidated;
```

File: src/Rule/CallbackHandler.php [hunk 4]
```diff
@@ -27,7 +27,7 @@ public function validate(mixed $value, object $rule, ValidationContext $context)
             throw new InvalidArgumentException('Using method outside of attribute scope is prohibited.');
         }
 
-        $result = $callback($value, $rule, $context);
+        $result = $callback($rule->getObjectBeingValidated() ?? $value, $rule, $context);
```
Comment (samdark) [inline src/Rule/CallbackHandler.php:30]:
```suggestion
        $result = $callback($rule->getObjectValidated() ?? $value, $rule, $context);
```

**Review by samdark — CHANGES_REQUESTED** [review]

**Comment by vjik** [comment]
@samdark I'm renamed the suggested properties/methods and rename same properties/merthods in `Count` rule.

**Review by samdark — APPROVED** [review]

**Pull request merged**